Java/jni/native_glue.cc

```diff
@@ -84,6 +89,68 @@ namespace litecore {
                 free((void *) _slice.buf);        // detached
         }
 
+        void jstringSlice::fixBrokenModifiedUTF8(const char *input){
+            // https://github.com/android-ndk/ndk/issues/283
+            size_t len = strlen(input);
+            size_t newLen = len;
+            queue<uint8_t*> modified;
+            const auto unsignedInput = (const uint8_t *)input;
+
+            // Need to figure out the actual length since each
+            // 4-bytes of real UTF-8 is 6 bytes of modified UTF-8
+            // but convert the bytes while we are at it
+            for(size_t i = 0; i < len; i++) {
```

Comment:
This function scans through the string twice. You can eliminate the `strlen` call by just changing the middle clause of the `for` to `unsignedInput[i] != 0`.

(I'm being picky about performance because it looks like this gets called on every single Java string passed into LiteCore.)